homeassistant/components/yamaha_musiccast/__init__.py

```diff
@@ -190,3 +199,36 @@ def device_info(self) -> DeviceInfo:
             device_info["via_device"] = (DOMAIN, self.coordinator.data.device_id)
 
         return device_info
+
+
+class MusicCastCapabilityEntity(MusicCastDeviceEntity):
+    """Base Entity type for all capabilities."""
+
+    def __init__(
+        self,
+        coordinator: MusicCastDataUpdateCoordinator,
+        capability: Capability,
+        zone_id: str = None,
+    ) -> None:
+        """Initialize a capability based entity."""
+        if zone_id is not None:
+            self._zone_id = zone_id
+        self.capability = capability
+        super().__init__(name=capability.name, icon="", coordinator=coordinator)
+        self._attr_entity_category = ENTITY_CATEGORY_MAPPING.get(capability.entity_type)
+
+    async def async_added_to_hass(self):
+        """Run when this Entity has been added to HA."""
+        await super().async_added_to_hass()
+        # All capability based entities should register callbacks to update HA when their state changes
+        self.coordinator.musiccast.register_callback(self.async_write_ha_state)
```

Comment:
In a future PR if `register_callback` returned a callback that canceled the registration you could do `self.async_on_remove(self.coordinator.musiccast.register_callback(self.async_write_ha_state))`